### ISSUE TITLE
[iOS] Hidden status bar not respected.

### ISSUE THREAD

**Issue by dustinbahr** [react-native-community/react-native-safe-area-view#62]
This SafeAreaView does not respect hidden status bar in iOS as it should.

Even, when status bar is hidden, there is a top inset. You can use `forceInsets` to override this, however, it will then ALSO override the inset on iPhones with a notch. (iPhone X, etc)

In some circumstances, you want the safe area to inset for the notch, even if the status bar info is hidden.

The SafeAreaView built in to react-native DOES treat it correctly. 

When status bar is hidden, there is still an inset on iPhone X, but not on normal iPhones.

Here are some examples:

**react-native-safe-area-view SafeAreaView**
```
import React from 'react';
import { View, StatusBar, Text } from 'react-native';
import SafeAreaView from 'react-native-safe-area-view';

export default class App extends React.Component {
  render() {
    return (
      <SafeAreaView style={{ backgroundColor: 'red', flex: 1 }}>
        <StatusBar hidden={true} />
        <View
          style={{
            backgroundColor: 'white',
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center'
          }}
        >
          <Text>SafeAreaView</Text>
          <Text>react-native-safe-area-view</Text>
        </View>
      </SafeAreaView>
    );
  }
}
```
![image](https://user-images.githubusercontent.com/1706540/54703481-80872580-4afe-11e9-87c7-c7bd9e20502d.png)

**react-native SafeAreaView**
```
import React from 'react';
import { View, StatusBar, Text, SafeAreaView } from 'react-native';

export default class App extends React.Component {
  render() {
    return (
      <SafeAreaView style={{ backgroundColor: 'red', flex: 1 }}>
        <StatusBar hidden={true} />
        <View
          style={{
            backgroundColor: 'white',
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center'
          }}
        >
          <Text>SafeAreaView</Text>
          <Text>react-native</Text>
        </View>
      </SafeAreaView>
    );
  }
}
```
![image](https://user-images.githubusercontent.com/1706540/54703573-aad8e300-4afe-11e9-8e42-43ac884b8961.png)

**Comment by dustinbahr** [comment]
I've created a PR to address this, although I feel like there should be a better way. #63

**Comment by brentvatne** [comment]
I agree that there should be a better way. I'm very much looking forward to this landing: https://github.com/facebook/react-native/pull/20999

**Comment by brentvatne** [comment]
this should be resolved in version 1.0 which was released today, thanks for reporting!
